doc: update known issues related to modemlib (#8810)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

Other issues
============

.. rst-class:: v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

> NCSDK-14312: During execution, in rare cases, the :c:func:`nrf_recv()` function would crash.

Reviewer: nit: This is still present for these versions.
```suggestion
NCSDK-14312: During execution, in rare cases, the :c:func:`nrf_recv()` function crash.
```

Author: The wording is correct since "would" implies that it happens occasionally.

> .. rst-class:: v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0
> 
> NCSDK-14312: During execution, in rare cases, the :c:func:`nrf_recv()` function would crash.
> 
> .. rst-class:: v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0
> 
> NCSDK-13360: When a socket that is in a :c:func:`nrf_recv()` loop is closed by another thread, it will eventually access the packet handler in the socket context causing a crash since the socket context was destroyed by the :c:func:`nrf_close()` call.

Reviewer 2: ```suggestion
.. rst-class:: v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-14312: The :c:func:`nrf_recv` function crashes
  During execution, in rare cases, the :c:func:`nrf_recv` function crashes  in the :ref:`nrfxlib:nrf_modem`.

.. rst-class:: v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-13360: The socket context crashes when closed by another thread
  When a socket that is in a :c:func:`nrf_recv` function loop is closed by another thread, the socket eventually accesses the packet handler in the socket context, causing a crash.
  This is because the socket context was destroyed by the :c:func:`nrf_close` function call.
```
1. We could move both the issues to nrfxlib, create a new section for "Modem Library" after the "GNSS sockets" header.
2. Just re-confirming whether we need to add v1-6 because the issue will now be fixed in this ncs version?

Reviewer 3: There is no tag on the v1.6-branch where these bugs are fixed (yet?). Both v1.6.0 and v1.6.1 have this bug.

> NCSDK-13360: When a socket that is in a :c:func:`nrf_recv()` loop is closed by another thread, it will eventually access the packet handler in the socket context causing a crash since the socket context was destroyed by the :c:func:`nrf_close()` call.

Reviewer 3: `The socket context crashes when closed by another thread` -> `Closing a socket while another thread is receiving data from it may crash`

"When a socket that is in a nrf_recv() function ~~loop~~", it does not need to be a loop, it just needs to be in a recv() call

  Moreover, closing a socket does not properly clean up all memory resources.
  If a socket is opened and closed multiple times, this  might starve the system.

Modem Library
============

.. rst-class:: v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-14312: The :c:func:`nrf_recv` crashes occasionally
  During execution, in rare cases, the :c:func:`nrf_recv` function crashes.

.. rst-class:: v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

> NCSDK-13360: When calling :c:func:`nrf_recv`, the socket context crashes if closed by another thread while receiving data

Reviewer: nit:
```suggestion
NCSDK-13360: When calling :c:func:`nrf_recv`, the socket context crashes if closed by another thread while receiving data.
```

Author: First line shouldn't have a full stop, but as far as I can see, the layout is one brief line to describe the problem and then a full description of the problem (with full stop). I shall update it again to align it to this layout.

  Moreover, closing a socket does not properly clean up all memory resources.
  If a socket is opened and closed multiple times, this  might starve the system.

> Modem Library
> ============

Reviewer 2: ```suggestion
Modem Library
=============
```

.. rst-class:: v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-14312: The :c:func:`nrf_recv` function crashes occasionally
  During execution, in rare cases, the :c:func:`nrf_recv` function crashes because of a race condition.

.. rst-class:: v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

> NCSDK-13360: The socket context crashes if closed by another thread while receiving data

Reviewer 3: "socket context" does not make much sense, what crashes is the `nrf_recv()` function call

>   When calling the :c:func:`nrf_recv` function, closing the socket from another thread while it's receiving data causes a crash.

Reviewer 2: ```suggestion
  When calling the :c:func:`nrf_recv` function, closing the socket from another thread while it is receiving data causes a crash.
```
